packages/list/ListCommand.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+var annotationHelpers = require('../../model/annotationHelpers');
+var SurfaceCommand = require('../../ui/SurfaceCommand');
+var uuid = require('../../util/uuid');
+
+
+var ListCommand = function(surface) {
+  SurfaceCommand.call(this, surface);
+};
+
+ListCommand.Prototype = function() {
+
+  this.getSelection = function() {
+    return this.getSurface().getSelection();
+  };
+
+  this.getCommandState = function() {
+    var surface = this.getSurface();
+    var sel = this.getSelection();
+    var disabled = !surface.isEnabled() || sel.isNull() || !sel.isPropertySelection();
+
+    return {
+      active: true,
+      disabled: disabled
+    };
+  };
+
+  // Execute command and trigger transformations
+  // TODO: This currently works for preoperty selection only. Make it work on container selection too.
+  this.execute = function() {
+    var sel = this.getSelection();
+    if (!sel.isPropertySelection()) return;
+    var doc = this.getDocument();
+    var surface = this.getSurface();
+
+    var path = sel.getPath();
+    var node = doc.get(path[0]);
+    var content = node.content;
+    var containerId = this.getContainerId();
+
+    surface.transaction(function(tx, args) {
+      var newList;
+      var container = tx.get(containerId);
+      if (node.type === 'list-item') {
```

Comment:
Splitting it out into two separate methods, one for text-> list and other for list -> text will make it clearer.